kore/src/Kore/Internal/TermLike/TermLike.hs

```diff
@@ -434,6 +496,190 @@ instance From (KeyF child) (TermLikeF variable child) where
     from (Key.StringLiteralF stringLiteral) = StringLiteralF stringLiteral
     {-# INLINE from #-}
 
+-- | @TermAttributes@ are the attributes of a pattern collected during verification.
+data TermAttributes variable = TermAttributes
+    { termSort :: !Sort
+    , termFreeVariables :: !(Attribute.FreeVariables variable)
+    , termFunctional :: !Attribute.Functional
+    , termFunction :: !Attribute.Function
+    , termDefined :: !Attribute.Defined
+    , termCreated :: !Attribute.Created
+    , termSimplified :: !Attribute.Simplified
+    , termConstructorLike :: !Attribute.ConstructorLike
+    , termSubterms :: !(Subterms variable)
+    }
+    deriving stock (Eq, Show)
+    deriving stock (GHC.Generic)
+    deriving anyclass (Hashable, NFData)
+    deriving anyclass (SOP.Generic, SOP.HasDatatypeInfo)
+
+instance Debug variable => Debug (TermAttributes variable) where
+    debugPrecBrief _ _ = "_"
+
+instance (Debug variable, Diff variable) => Diff (TermAttributes variable)
+
+instance
+    ( Synthetic Sort base
+    , Synthetic (Attribute.FreeVariables variable) base
+    , Synthetic Attribute.Functional base
+    , Synthetic Attribute.Function base
+    , Synthetic Attribute.Defined base
+    , Synthetic Attribute.Simplified base
+    , Synthetic Attribute.ConstructorLike base
+    , Synthetic (Subterms variable) base
+    ) =>
+    Synthetic (TermAttributes variable) base
+    where
+    synthetic base =
+        TermAttributes
+            { termSort = synthetic (termSort <$> base)
+            , termFreeVariables = synthetic (termFreeVariables <$> base)
+            , termFunctional = synthetic (termFunctional <$> base)
+            , termFunction = synthetic (termFunction <$> base)
+            , termDefined = synthetic (termDefined <$> base)
+            , termCreated = synthetic (termCreated <$> base)
+            , termSimplified =
+                if Attribute.isConstructorLike constructorLikeAttr
+                    then Attribute.fullySimplified
+                    else synthetic (termSimplified <$> base)
+            , termConstructorLike = constructorLikeAttr
+            , termSubterms = synthetic (termSubterms <$> base)
+            }
```

Comment:
```suggestion
    synthetic base =
        let ~attrs = TermAttributes
            { termSort = synthetic (termSort <$> base)
            , termFreeVariables = synthetic (termFreeVariables <$> base)
            , termFunctional = synthetic (termFunctional <$> base)
            , termFunction = synthetic (termFunction <$> base)
            , termDefined = synthetic (termDefined <$> base)
            , termCreated = synthetic (termCreated <$> base)
            , termSimplified =
                if Attribute.isConstructorLike constructorLikeAttr
                    then Attribute.fullySimplified
                    else synthetic (termSimplified <$> base)
            , termConstructorLike = constructorLikeAttr
            , termSubterms = subterms
            }
            self = Recursive.embed (attrs :< base)
            subterms = HashSet.insert self (foldMap termSubterms base)
```

Reply:
@ttuegel I'm having trouble implementing this as you suggested. I think the issue comes from `base` being of type `base (TermAttributes variable)` instead of `base (TermLike variable)`, meaning I can't construct the `TermLike` in `self`.